src/ecdsa/recovery.rs

```diff
@@ -192,6 +192,19 @@ impl<C: Verification> Secp256k1<C> {
         msg: &Message,
         sig: &RecoverableSignature,
     ) -> Result<key::PublicKey, Error> {
+        cfg_if::cfg_if! {
+            if #[cfg(all(target_os = "zkvm", target_vendor = "succinct"))] {
+                let prehash: &[u8] = &msg.0;
+                let mut rev_sig = sig.0[..32].iter().rev().chain(sig.0[32..64].iter().rev()).cloned().collect::<Vec<u8>>();
```

Comment:
Specify the endian-ness of the incoming data and make the logic easy to follow with comments.

```rust
let sig_bytes_be = sig.0
# In XYZ, it's LE, need to convert to this.
let sig_bytes_le = 
```

Reply:
Explained